File: Java/14.longest.common.prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }
        
        // Sort the array of strings
        Arrays.sort(strs);
        
        // Take the first and last strings after sorting
        String first = strs[0];
        String last = strs[strs.length - 1];
        
        // Find the common prefix between the first and last strings
        StringBuilder commonPrefix = new StringBuilder();
        for (int i = 0; i < first.length() && i < last.length(); i++) {
            if (first.charAt(i) == last.charAt(i)) {
                commonPrefix.append(first.charAt(i));
            } else {
                break;
            }
        }
        
        return commonPrefix.toString();
    }
}

File: C#/14.longest.common.prefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        if (strs == null || strs.Length == 0) {
            return "";
        }

        // Sort the array of strings
        Array.Sort(strs);

        // Take the first and last strings after sorting
        string first = strs[0];
        string last = strs[strs.Length - 1];

        // Find the common prefix between the first and last strings
        StringBuilder commonPrefix = new StringBuilder();
        for (int i = 0; i < first.Length && i < last.Length; i++) {
            if (first[i] == last[i]) {
                commonPrefix.Append(first[i]);
            } else {
                break;
            }
        }

        return commonPrefix.ToString();
    }
}